packages/flutter/lib/src/cupertino/cupertino_focus_halo.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'constants.dart';
+
+/// {@template flutter.cupertino.CupertinoFocusHalo}
+/// Applies a Cupertino-style focus border around its child when any of child focus nodes gain focus.
+///
+/// See also:
+///
+/// * <https://developer.apple.com/design/human-interface-guidelines/focus-and-selection/>
+/// {@endtemplate}
+class CupertinoFocusHalo extends StatefulWidget {
+  /// {@macro flutter.cupertino.CupertinoFocusHalo}
```

Comment:
Typically we don't repeat the class docs on each constructor. I think you can just add this in place of the macro: "Creates a rectangular [CupertinoFocusHalo] around the child." Then users can click the link to the class docs if they're curious.